webapp/src/shared/components/DispatchPage/DispatchPage.tsx

```diff
@@ -20,7 +30,33 @@ export default function DispatchPage<T>({
   dataMapper,
   button,
 }: DispatchPageProps<T>) {
-  const [data, setData] = useState<T[]>([]);
+  const [search, setSearch] = useState('');
```

Comment:
since we already have the type, I think something like this may make it a bit cleaner:
```
diff --git a/webapp/src/shared/components/DispatchPage/DispatchPage.tsx b/webapp/src/shared/components/DispatchPage/DispatchPage.tsx
index 30d284d..3eaecf9 100644
--- a/webapp/src/shared/components/DispatchPage/DispatchPage.tsx
+++ b/webapp/src/shared/components/DispatchPage/DispatchPage.tsx
@@ -30,8 +30,11 @@ export default function DispatchPage<T>({
   dataMapper,
   button,
 }: DispatchPageProps<T>) {
-  const [search, setSearch] = useState('');
-  const [offset, setOffset] = useState(0);
+  const [searchParams, setSearchParams] = useState<SearchFetchFnParams>({
+    search: '',
+    offset: 0,
+  });
+  const { search, offset } = searchParams;
   const getData = useCallback(() => {
     return fetchFn({ search, offset });
   }, [fetchFn, search, offset]);
@@ -48,7 +51,7 @@ export default function DispatchPage<T>({
       }
       observer.current = new IntersectionObserver((entries) => {
         if (entries[0].isIntersecting && hasMore) {
-          setOffset(data.length);
+          setSearchParams({ ...searchParams, offset: data.length });
         }
       });
       if (row) {
@@ -76,8 +79,7 @@ export default function DispatchPage<T>({
   };
 
   const handleSearch = (value: string) => {
-    setSearch(value);
-    setOffset(0);
+    setSearchParams({ search: value, offset: 0 });
   };
 
   return (
```

Reply:
Thank you. Done